internal/databases/mysql/mysql.go

```diff
@@ -243,6 +254,71 @@ outer:
 	return nil
 }
 
+func provideLogs(folder storage.Folder, dstDir string, startTS, endTS time.Time, p *storage.ObjectProvider) {
+	defer p.Close()
+	_, err := os.Stat(dstDir)
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(dstDir, 0700)
+		if err != nil {
+			ok := p.AddErrorToProvider(err)
+			for !ok {
+				ok = p.AddErrorToProvider(err)
+			}
```

Comment:
Let's hide error handling logic, like HandleEventError in binlog_server_handler.go